### PR TITLE
irrlicht: 1.8.4 -> 1.8.5

### LINKED ISSUE #321031: Update request: `irrlicht`  1.8.4 → 1.8.5
- Package name: `irrlicht`
- Latest released version:

- Current version on the unstable channel: 1.8.4
- Current version on the stable/release channel: 1.8.4

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls) − No PR opened

**Notify maintainers**

@danielfullmer (authored #28711)
@astro (authored #141219)

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes
Fixes #321031

Changelog: https://irrlicht.sourceforge.io/forum/viewtopic.php?f=2&t=52775

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pinpox** [comment]
Result of `nixpkgs-review pr 321139` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>irrlicht</li>
    <li>voxelands</li>
  </ul>
</details>

**Review by pinpox — APPROVED** [review]

**Comment by pbsds** [comment]
https://hydra.nixos.org/build/265430711
https://hydra.nixos.org/build/265390839
please mark broken on darwin

File: pkgs/development/libraries/irrlicht/mac.nix
Comment (pbsds) [inline pkgs/development/libraries/irrlicht/mac.nix:42]:
```suggestion
    broken = stdenv.isDarwin;
```

or 

```suggestion
    badPlatforms = lib.platforms.darwin;
```

**Comment by pbsds** [comment]
Hi, why close this PR?

**Comment by pyrox0** [comment]
I do not have a darwin machine, and thus I do not want to keep trying to fix darwin build failures. If you would like to take the work here and make it work on darwin, feel free, but I will not be doing that myself.

**Comment by pbsds** [comment]
I neither have darwin, and consider the slow darwin CI a major obstacle to nixpkgs. The fact that i have to leave PRs alone for several days before safely merging them simply infuriates me. This is why i am so quick to suggest marking darwin broken and leaving it for others to solve.

The final suggestion i came with is to properly stop darwin builds. `meta.broken` expects a boolean, not a list of system strings. `meta.badPlatforms` on the other hand accepts a list of system strings. Yours was thing in the middle, hence why i suggested both options.

**Comment by pyrox0** [comment]
Again, feel free to submit a pr yourself using this code. I'm not saying don't merge, I'm saying that I will not deal with darwin problems when I have enough to deal with elsewhere in nixpkgs and my life. Feel free to pick this up or mark it as broken, or don't. I was doing this since I was interested in updating it, but I have no more interest in doing so, so I've closed the PR.

**Pull request closed without merging**
